### PR TITLE
fix: include dot-files support in glob selection

### PR DESCRIPTION
**Pull request by benPearce1**
[sc-40495]

Using the following action yaml

```yaml
      - name: Run build script
        run: | 
          mkdir work
          cd work
          echo "test output" > .dotfile
          cat .dotfile
          touch test.txt
          touch no-extension
        shell: bash
      - name: Create Zip Package for Octopus Deploy
      # You may pin to the exact commit or the version.
        uses: OctopusDeploy/create-zip-package-action@06107900912e6dc6be9a56f4340cd54a8024c61a
        with:
          package_id: output
          version: 1.0.0
          output_folder: ./pack
          base_path: .
          files: |
            work/**/*
```

produces a zip file with the following contents:
image

Also, added a developers guide

### CODE REVIEW

File: DEVELOPERS_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+```
+npm run build
+git add dist/.
+git commit -m "updating index.js"
+git log -q -n 1 dist/index.js | less -F
+```
+
+From the log output take note of the commit hash and push to GitHub
+
+In a test GitHub action you can use the branched build of the action by referencing the commit hash
```
Comment (slewis74) [inline DEVELOPERS_GUIDE.md:14]:
```suggestion
In a test GitHub action you can use the branched build of the action by referencing the branch or commit hash, see [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses) for details on the `uses` syntax.
```

File: DEVELOPERS_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+```
+- name: Create Zip Package for Octopus Deploy
+      # You may pin to the exact commit or the version.
+        uses: OctopusDeploy/create-zip-package-action@6d73e1f62030dfa23534ff0d279774177d5ea1ea
```
Comment (slewis74) [inline DEVELOPERS_GUIDE.md:19]:
```suggestion
        uses: OctopusDeploy/create-zip-package-action@my-branch
```
I've found it's much easier to work with just the branch name, not specific commits. You can run the workflow in the consuming repo then without requiring an update to the commit hash as you make changes.

File: DEVELOPERS_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          package_id: output
+          version: 1.0.0
+          output_folder: ./pack
+          base_path: .
+          files: |
+            **/*
+            **/.*
```
Comment (slewis74) [inline DEVELOPERS_GUIDE.md:27]:
```suggestion

```
Is the `**/*` example enough?

**Review by slewis74 — APPROVED** [review]
couple of suggestions for the dev guide

File: DEVELOPERS_GUIDE.md (hunk 3 above)
Comment (benPearce1) [reply DEVELOPERS_GUIDE.md:27]: yeah, this was just a copy paste

File: DEVELOPERS_GUIDE.md (hunk 2 above)
Comment (benPearce1) [reply DEVELOPERS_GUIDE.md:19]: good point

**Pull request merged**